lib/init/features/firestore.js

```diff
@@ -27,11 +34,62 @@ var _initRules = function(setup, config) {
       type: "input",
       name: "rules",
       message: "What file should be used for Firestore Rules?",
-      default: "firestore.rules",
+      default: DEFAULT_RULES_FILE,
     },
-  ]).then(function() {
-    return config.writeProjectFile(setup.config.firestore.rules, RULES_TEMPLATE);
-  });
+  ])
+    .then(function() {
+      var filename = setup.config.firestore.rules;
+
+      if (fsutils.fileExistsSync(filename)) {
+        var msg =
+          "File " +
+          chalk.bold(filename) +
+          " already exists." +
+          " Do you want to overwrite it with the Firestore Rules from the Firebase Console?";
+        return prompt.once({
+          type: "confirm",
+          message: msg,
+          default: false,
+        });
+      }
+
+      return Promise.resolve(true);
+    })
+    .then(function(overwrite) {
+      if (!overwrite) {
+        return Promise.resolve();
+      }
+
+      var projectId = setup.instance || setup.rcfile.projects.default;
+      return _getRulesFromConsole(projectId).then(function(contents) {
+        return config.writeProjectFile(setup.config.firestore.rules, contents);
+      });
+    });
+};
+
+var _getRulesFromConsole = function(projectId) {
+  return gcp.rules
+    .getLatestRulesetName(projectId)
+    .then(function(name) {
+      if (!name) {
+        logger.debug("No rulesets found, using default.");
+        var defaultContent = [{ name: DEFAULT_RULES_FILE, content: RULES_TEMPLATE }];
+        return Promise.resolve(defaultContent);
```

Comment:
No need for "Promise.resolve", since any returns in a "then" already gets wrapped in a promise.